### PR TITLE
Chore: Various updates

### PR DESCRIPTION
**Pull request by Elfo404**
- updates OTel & tooling to 0.96.0
- removes go.work.sum. should only live locally and not be pushed
- replace deprecate `ReportFatalError` in the drone receiver with `ReportStatus`
- adds github actions receiver to the distro (our own version with logs support)
- removes `toolchain` from each go.mod 
- updates bingo to 0.9
- update go to 1.22
- update docker image to `golang:1.22.1-alpine3.19`
- Refactored a bit drone receiver to be more in line with GHA

## Note

The config slighlty changed, the webhook now needs to be configured as follows:

```yaml
receivers:
  dronereceiver:
    path: /drone/webhook
    endpoint: localhost:3333
    secret: mysecret
```

### CODE REVIEW

File: pkg/dronereceiver/factory.go [hunk 1]
```diff
@@ -2,53 +2,68 @@ package dronereceiver
...
+
+	if consumer == nil {
+		return nil, component.ErrNilNextConsumer
+	}
+
+	r := receivers.GetOrAdd(cfg, func() component.Component {
+		var rcv component.Component
+		rcv, err = newReceiver(set, rCfg)
+		return rcv
+	})
```
Comment (dsotirakis) [inline pkg/dronereceiver/factory.go:54]: wondering whether we should check for `r` value being nil here

File: pkg/dronereceiver/factory.go [hunk 2]
```diff
@@ -2,53 +2,68 @@ package dronereceiver
...
 }
 
-func createMetricsReceiver(_ context.Context, set receiver.CreateSettings, rConf component.Config, consumer consumer.Metrics) (receiver.Metrics, error) {
+func newMetricsReceiver(_ context.Context, set receiver.CreateSettings, rConf component.Config, consumer consumer.Metrics) (receiver.Metrics, error) {
 	cfg := rConf.(*Config)
 	ns := newDroneScraper(set, cfg)
-	scraper, err := scraperhelper.NewScraper(metadata.Type, ns.scrape, scraperhelper.WithStart(ns.start))
+	scraper, err := scraperhelper.NewScraper(string(metadata.Type), ns.scrape, scraperhelper.WithStart(ns.start))
 
 	if err != nil {
```
Comment (dsotirakis) [inline pkg/dronereceiver/factory.go:68]: should we check if the consumer is empty here as well?

File: pkg/dronereceiver/handler.go [hunk 3]
```diff
@@ -67,24 +37,22 @@ func (d *droneWebhookHandler) handler(resp http.ResponseWriter, req *http.Reques
 	// However this appears to be sent when a build completes; The structure however changes slightly as in
 	// the `repo.build` seems to be absent in the first one.
 	// TODO: Revisit this, we may not need the Finished check.
-	if evt.Repo.Build == nil || evt.Repo.Build.Finished == 0 {
-		return fmt.Errorf("no build info provided from the webhook event: %v", err)
+	if build == nil || build.Finished == 0 {
+		logger.Error("no build info provided from the webhook event")
```
Comment (dsotirakis) [inline pkg/dronereceiver/handler.go:41]: should we error here? I guess we could Warn?

File: pkg/dronereceiver/handler.go [hunk 4]
```diff
@@ -67,24 +37,22 @@ func (d *droneWebhookHandler) handler(resp http.ResponseWriter, req *http.Reques
...
-	repo := evt.Repo
-	build := evt.Repo.Build
-
 	// Skip traces for repos that are not enabled
-	allowedBranches, ok := d.reposConfig[evt.Repo.Slug]
+	allowedBranches, ok := config.ReposConfig[evt.Repo.Slug]
 	if !ok {
-		d.logger.Warn("repo not enabled", zap.String("repo", evt.Repo.Slug))
-		return nil
+		logger.Info("repo not enabled, skipping", zap.String("repo", evt.Repo.Slug))
```
Comment (dsotirakis) [inline pkg/dronereceiver/handler.go:48]: maybe also Warn?

**Review by dsotirakis — APPROVED** [review]
nits - great work 💪

File: pkg/dronereceiver/factory.go (hunk 1 above)
Comment (Elfo404) [reply pkg/dronereceiver/factory.go:54]: if `err` is not nil `r` cannot possibly be nil (according to `newReceiver` implementation). maybe we can add more coverage for `newReceiver` to check that is actually true (now we are only testing the happy path) I personally wouldn't introduce more code that cannot really be reached, i'd rather write more test to validate and lock the behavior. wdyt?

Comment (dsotirakis) [reply pkg/dronereceiver/factory.go:54]: yeah that sounds good!

File: pkg/dronereceiver/factory.go (hunk 2 above)
Comment (Elfo404) [reply pkg/dronereceiver/factory.go:68]: the plan was to remove metrics from the receiver anyway in a follow up PR (to be more in line with the direction other CI/CD related receivers are going) so i didn't really touch anything around metrics, but yep, makes sense. will change 👍

File: pkg/dronereceiver/handler.go (hunk 4 above)
Comment (Elfo404) [reply pkg/dronereceiver/handler.go]: done in 96d49d6

File: pkg/dronereceiver/handler.go (hunk 3 above)
Comment (Elfo404) [reply pkg/dronereceiver/handler.go]: done in 96d49d6

File: pkg/dronereceiver/factory.go (hunk 2 above)
Comment (Elfo404) [reply pkg/dronereceiver/factory.go:73]: done in b2bb8d6

**Pull request merged**